devbox.json

```
@@ -1,45 +1,55 @@
   },
   "env": {
     "GOENV": "off",
-    "PATH":  "$PATH:$PWD/dist",
+    "PATH": "$PATH:$PWD/dist"
   },
   "shell": {
     "init_hook": [
       // Remove Go environment variables that might've been inherited from the
       // user's environment and could affect the build.
       "test -z $FISH_VERSION && \\",
       "unset       CGO_ENABLED GO111MODULE GOARCH GOFLAGS GOMOD GOOS GOROOT GOTOOLCHAIN GOWORK || \\",
-      "set --erase CGO_ENABLED GO111MODULE GOARCH GOFLAGS GOMOD GOOS GOROOT GOTOOLCHAIN GOWORK",
+      "set --erase CGO_ENABLED GO111MODULE GOARCH GOFLAGS GOMOD GOOS GOROOT GOTOOLCHAIN GOWORK"
     ],
     "scripts": {
       // Build devbox for the current platform
-      "build":              "go build -o dist/devbox ./cmd/devbox",
+      "build": "go build -o dist/devbox ./cmd/devbox",
       "build-darwin-amd64": "GOOS=darwin GOARCH=amd64 go build -o dist/devbox-darwin-amd64 ./cmd/devbox",
       "build-darwin-arm64": "GOOS=darwin GOARCH=arm64 go build -o dist/devbox-darwin-arm64 ./cmd/devbox",
-      "build-linux-amd64":  "GOOS=linux GOARCH=amd64 go build -o dist/devbox-linux-amd64 ./cmd/devbox",
-      "build-linux-arm64":  "GOOS=linux GOARCH=arm64 go build -o dist/devbox-linux-arm64 ./cmd/devbox",
+      "build-linux-amd64": "GOOS=linux GOARCH=amd64 go build -o dist/devbox-linux-amd64 ./cmd/devbox",
+      "build-linux-arm64": "GOOS=linux GOARCH=arm64 go build -o dist/devbox-linux-arm64 ./cmd/devbox",
       "build-all": [
         "devbox run build-darwin-amd64",
         "devbox run build-darwin-arm64",
         "devbox run build-linux-amd64",
-        "devbox run build-linux-arm64",
+        "devbox run build-linux-arm64"
       ],
       // Open VSCode
-      "code":               "code .",
-      "lint":               "golangci-lint run --timeout 5m && scripts/gofumpt.sh",
-      "fmt":                "scripts/gofumpt.sh",
-      "test":               "go test -race -cover ./...",
+      "code": "code .",
+      "lint": "golangci-lint run --timeout 5m && scripts/gofumpt.sh",
+      "fmt": "scripts/gofumpt.sh",
+      "test": "go test -race -cover ./...",
       "test-projects-only": "DEVBOX_RUN_PROJECT_TESTS=1 go test -v -timeout ${DEVBOX_GOLANG_TEST_TIMEOUT:-30m} ./... -run \"TestExamples|TestScriptsWithProjects\"",
-      "update-examples":    "devbox run build && go run testscripts/testrunner/updater/main.go",
-      "tidy":               "go mod tidy",
-
+      "update-examples": "devbox run build && go run testscripts/testrunner/updater/main.go",
+      // Updates the Flake's vendorHash: First run `go mod vendor` to vendor
+      // the dependencies, then hash the vendor directory with Nix.
+      // The hash is saved to the `vendor-hash` file, which is then
+      // read into the Nix Flake.
+      "update-hash": [
+        "go mod vendor",
+        "hash=$(nix hash path vendor)",
+        "echo \"$hash\" > vendor-hash",
+        "rm -rf vendor",
+        "echo \"Vendor Hash: $hash\""
```

Could we just vendor into a temp dir? I'm not sure what something like gopls will do if it suddenly sees a new vendor directory.

```suggestion
        // realpath to work-around nix hash not liking symlinks
        "vendor=$(realpath $(mktemp -d))",
        "trap \"rm -rf $vendor\" EXIT",
        "go mod vendor -o $vendor",
        "nix hash path $vendor >vendor-hash",
```